### PR TITLE
Change flag for Jacobian generation

### PR DESCRIPTION
**Pull request by phannebohm**
### Related Issues

#10232

### Purpose

`--generateSymbolicJacobian` is replaced by `--generateDynamicJacobian` with options:
- `none`: Does not generate Jacobian.
- `numeric`: Generates sparsity pattern for numeric Jacobian.
- `symbolic`: Generates symbolic Jacobian. Default is `numeric` in order to mimic the old behavior.

### CODE REVIEW

File: OMCompiler/Compiler/BackEnd/SymbolicJacobian.mo [hunk 1]
```diff
@@ -94,11 +94,11 @@ public constant String DAE_CJ = "$DAE_CJ";
 
 public function symbolicJacobian "author: lochel
```
Comment (casella) [inline OMCompiler/Compiler/BackEnd/SymbolicJacobian.mo:95]: Do you want to also change the name of these functions? At least, if they work for both ODE and DAE mode, their documentation should say that clearly

Comment (casella) [reply OMCompiler/Compiler/BackEnd/SymbolicJacobian.mo:95]: If you search for "ODE" in this file, there are many instances of that...

File: OMCompiler/Compiler/Util/Flags.mo [hunk 2]
```diff
@@ -768,121 +746,93 @@ constant ConfigFlag POST_OPT_MODULES = CONFIG_FLAG(16, "postOptModules",
     ("solveLinearSystem", Gettext.notrans("solve linear system with newton step")),
     ("solveSimpleEquations", Gettext.notrans("Solves simple equations")),
     ("symSolver", Gettext.notrans("Rewrites the ode system for implicit Euler method. This module requires +symSolver.")),
-    ("symbolicJacobian", Gettext.notrans("Detects the sparse pattern of the ODE system and calculates also the symbolic Jacobian if flag '--generateSymbolicJacobian' is enabled.")),
+    ("symbolicJacobian", Gettext.notrans("Detects the sparse pattern of the ODE system and calculates also the symbolic Jacobian if flag '--generateDynamicJacobian=symbolic'.")),
```
Comment (casella) [inline OMCompiler/Compiler/Util/Flags.mo:749]: Only ODE or also DAE?

Comment (casella) [reply OMCompiler/Compiler/Util/Flags.mo:749]: Also elsewhere in this file...

**Review by casella — COMMENTED** [review]
Please see detailed comments in the code

File: OMCompiler/Compiler/Util/Flags.mo (hunk 2 above)
Comment (phannebohm) [reply OMCompiler/Compiler/Util/Flags.mo:749]: Only ODE. There are special ones for DAE:
https://github.com/OpenModelica/OpenModelica/blob/1711781d8ab93da9e5ba1c8466b537da28db838b/OMCompiler/Compiler/Util/Flags.mo#L1248-L1263

File: OMCompiler/Compiler/BackEnd/SymbolicJacobian.mo (hunk 1 above)
Comment (phannebohm) [reply OMCompiler/Compiler/BackEnd/SymbolicJacobian.mo:95]: There is a dedicated function for DAEs:
https://github.com/OpenModelica/OpenModelica/blob/1711781d8ab93da9e5ba1c8466b537da28db838b/OMCompiler/Compiler/BackEnd/SymbolicJacobian.mo#L211

I really don't feel familiar enough with that part of the code to tell apart ODE and DAE stuff, so I am leaving it as it is. No harm done.

**Pull request merged**